### PR TITLE
Basic iOS introduction

### PR DESCRIPTION
**Pull request by andrewxhill**
Signed-off-by: andrew <andrewxhill@gmail.com>

Snippets paid off here for sure!

### CODE REVIEW

File: docs/develop/clients/ios.md [hunk 1]
```diff
@@ -1,3 +1,129 @@
...
+```cmd
+pod 'Textile'
+```
+
+This will install the latest version of the SDK into your app. 
+
+#### Carthage
+
+```cmd
+github "textileio/ios-textile"
```
Comment (asutula) [inline docs/develop/clients/ios.md]: Would be amazed if this works. Does it?

File: snippets/common.yml [hunk 2]
```diff
@@ -901,7 +901,7 @@ messages:
             NSError *error;
             [Textile.instance.messages add:@"12D3KooWMSvp3cisDyu88Smgub8vMcMYnwsk1KdmtwVbLip9fkGk" body:@"hello?" error:&error];
             if (error) {
-              // Do somehting with this error
+              // Do something with this error
```
Comment (asutula) [inline snippets/common.yml]: Ha thanks, copy/paste mess

File: snippets/common.yml [hunk 3]
```diff
@@ -937,6 +937,21 @@ files:
...
+      code: |
+            var error: NSError?
+            NSString *result = [Textile.instance.files data:hash error:&error];
+            [Textile.instance.files data:hash completion:^(NSString *result, NSError *error) {
+              if (error) {
+                // Do something with this error
+              } else {
+                  // Success!
+              }
+            }];
```
Comment (asutula) [inline snippets/common.yml]: Not really sure what all this is other than it's not Swift.

Comment (andrewxhill) [reply snippets/common.yml]: oh yeah, ha, meant to flag that one for you. copy paste from the method below it just to start it. no idea what the right way to use that is in swift

Comment (asutula) [reply snippets/common.yml]: Ok here's the obj c:
```
  NSString *hash = @"some file hash";
  NSError *error;
  NSString *base64 = [Textile.instance.files data:hash error:&error];
  if (error) {
    // Do something with this error
  } else {
    // Success!
  }
```

and the Swift:
```
    let hash = "some file hash"
    var error: NSError?
    let base64 = Textile.instance().files.data(hash, error: &error)
    if (error != nil) {
      // Do something with this error
    } else {
      // Success!
    }```

File: docs/develop/clients/ios.md (hunk 1 above)
Comment (andrewxhill) [reply docs/develop/clients/ios.md]: it works! not certain about dependencies though. might just make the explicit to be safe

File: snippets/common.yml (hunk 3 above)
Comment (asutula) [reply snippets/common.yml]: related to these docs, just opened https://github.com/textileio/go-textile/issues/784

File: docs/develop/clients/ios.md (hunk 1 above)
Comment (asutula) [reply docs/develop/clients/ios.md]: coooool. maybe add some "experimental feature" blurb under it.

**Comment by andrewxhill** [comment]
okay, should be gtg

**Pull request merged**